x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/index/engine/FollowingEngine.java

```diff
@@ -119,13 +118,15 @@ protected long generateSeqNoForOperationOnPrimary(final Operation operation) {
 
     @Override
     protected void advanceMaxSeqNoOfUpdatesOrDeletesOnPrimary(long seqNo) {
-        if (Assertions.ENABLED) {
```

Comment:
I wonder if we can keep the assertion for deletes and skip it for updates? You would need to split the method into two, but at least we then still have some of the assertion left.